Flaky or broken TestDeltaLakeCreateTableAsSelectCompatibility.testReplaceTableWithSchemaChangeOnCheckpoint on 9.1 LTS

https://github.com/trinodb/trino/actions/runs/6700193910/job/18206194746?pr=19564

```
tests               | 2023-10-31 06:37:50 INFO: FAILURE     /    io.trino.tests.product.deltalake.TestDeltaLakeCreateTableAsSelectCompatibility.testReplaceTableWithSchemaChangeOnCheckpoint (Groups: profile_specific_tests, delta-lake-databricks, delta-lake-exclude-133, delta-lake-oss) took 36.9 seconds
tests               | 2023-10-31 06:37:50 SEVERE: Failure cause:
tests               | java.lang.AssertionError: expected [00000000000000000011.json] but found [00000000000000000012.json]
tests               | 	at org.testng.Assert.fail(Assert.java:94)
tests               | 	at org.testng.Assert.failNotEquals(Assert.java:513)
tests               | 	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
tests               | 	at org.testng.Assert.assertEquals(Assert.java:116)
tests               | 	at org.testng.Assert.assertEquals(Assert.java:190)
tests               | 	at org.testng.Assert.assertEquals(Assert.java:200)
tests               | 	at io.trino.tests.product.deltalake.TransactionLogAssertions.assertTransactionLogVersion(TransactionLogAssertions.java:50)
tests               | 	at io.trino.tests.product.deltalake.TestDeltaLakeCreateTableAsSelectCompatibility.testReplaceTableWithSchemaChangeOnCheckpoint(TestDeltaLakeCreateTableAsSelectCompatibility.java:195)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCreateTableAsSelectCompatibility.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCreateTableAsSelectCompatibility.java
@@ -32,7 +32,6 @@ import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
 import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS_104;
 import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS_113;
 import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS_122;
-import static io.trino.tests.product.TestGroups.DELTA_LAKE_EXCLUDE_133;
 import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
 import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
 import static io.trino.tests.product.deltalake.TransactionLogAssertions.assertLastEntryIsCheckpointed;
@@ -172,9 +171,7 @@ public class TestDeltaLakeCreateTableAsSelectCompatibility
         }
     }
 
-    // Databricks 11.3, 12.2 and 13.3 don't create a checkpoint file at 'CREATE OR REPLACE TABLE' statement
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_133, PROFILE_SPECIFIC_TESTS})
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
     public void testReplaceTableWithSchemaChangeOnCheckpoint()
     {
         String tableName = "test_replace_table_with_schema_change_" + randomNameSuffix();

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TransactionLogAssertions.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TransactionLogAssertions.java
@@ -15,19 +15,15 @@ package io.trino.tests.product.deltalake;
 
 import com.amazonaws.services.s3.AmazonS3;
 import com.amazonaws.services.s3.model.ListObjectsV2Result;
-import com.google.common.base.Throwables;
-import org.testng.Assert.ThrowingRunnable;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Predicate;
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
-import static java.lang.Long.parseLong;
 import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.fail;
 
 public class TransactionLogAssertions
 {
@@ -48,43 +44,6 @@ public class TransactionLogAssertions
         Optional<String> lastJsonEntry = listJsonLogEntries(s3Client, bucketName, tableName).stream().max(String::compareTo);
         assertThat(lastJsonEntry).isPresent();
         assertEquals(lastJsonEntry.get(), format("%020d.json", versionNumber));
-    }
-
-    public static void assertNewVersion(AmazonS3 s3Client, String bucketName, String tableName, ThrowingRunnable runnable)
-    {
-        long initialVersion = getTransactionLogVersion(s3Client, bucketName, tableName);
-        try {
-            runnable.run();
-        }
-        catch (Throwable e) {
-            Throwables.throwIfUnchecked(e);
-            throw new RuntimeException(e);
-        }
-        assertThat(getTransactionLogVersion(s3Client, bucketName, tableName))
-                .isGreaterThan(initialVersion);
-    }
-
-    public static void assertNoNewVersion(AmazonS3 s3Client, String bucketName, String tableName, ThrowingRunnable runnable)
-    {
-        long initialVersion = getTransactionLogVersion(s3Client, bucketName, tableName);
-        try {
-            runnable.run();
-        }
-        catch (Throwable e) {
-            Throwables.throwIfUnchecked(e);
-            throw new RuntimeException(e);
-        }
-        assertThat(getTransactionLogVersion(s3Client, bucketName, tableName))
-                .isEqualTo(initialVersion);
-    }
-
-    private static long getTransactionLogVersion(AmazonS3 s3Client, String bucketName, String tableName)
-    {
-        Optional<String> lastJsonEntry = listJsonLogEntries(s3Client, bucketName, tableName).stream().max(String::compareTo);
-        if (lastJsonEntry.isEmpty()) {
-            fail("Cannot determine version for table " + tableName);
-        }
-        return parseLong(lastJsonEntry.get().split("\\.")[0]);
     }
 
     private static List<String> listJsonLogEntries(AmazonS3 s3Client, String bucketName, String tableName)

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TransactionLogAssertions.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TransactionLogAssertions.java
@@ -25,7 +25,7 @@ import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.testng.Assert.assertEquals;
 
-public class TransactionLogAssertions
+public final class TransactionLogAssertions
 {
     private TransactionLogAssertions() {}
 

